src/js/main.js

```diff
@@ -32,6 +33,163 @@ Site = {
   },
 };
 
+Site.Programacion = {
+  // parse JSON event posts object from Wordpress
+  eventsPosts: JSON.parse(WP.programacionEvents),
+  scheduleArray: [],
+  programacionContainer: document.getElementById('page-programacion'),
+
+  init: function() {
+    var _this = this;
+
+    // loop through Years -> Months -> Days -> Events
+    _this.fillEventsObject();
+  },
+
+  fillEventsObject: function() {
+    var _this = this;
+
+    // forEach event
+    _this.eventsPosts.forEach(function(el) {
+      // generate moment from timestamp in user local timezone
+      var eventMoment = moment.unix(parseInt(el.timestamp));
+
+      var year = eventMoment.format('YYYY');
+      var month = eventMoment.format('MMMM');
+      var monthNum = eventMoment.format('M');
+      var day = eventMoment.format('dddd Do');
+      var dayNum = eventMoment.format('D');
+      var hour = eventMoment.format('H:mm');
+
+      // Create this Year object if missing in schedule array
+      if (!_this.scheduleArray[year]) {
+        _this.scheduleArray[year] = [];
+        _this.scheduleArray[year] = {
+          year: year,
+          months: []
+        }
+      }
+
+      // Create this Month object if missing in Year
+      if (_this.scheduleArray[year].months[monthNum] === undefined) {
+        _this.scheduleArray[year].months[monthNum] = {
+          month: month,
+          days: []
+        };
+      }
+
+      // Create this Day object if missing in Month
+      if (_this.scheduleArray[year].months[monthNum].days[dayNum] === undefined) {
+        _this.scheduleArray[year].months[monthNum].days[dayNum] = {
+          day: day,
+          events: []
+        };
+      }
+
+      // Push Event object to events array
+      _this.scheduleArray[year].months[monthNum].days[dayNum].events.push({
+        hour: hour,
+        title: el.title,
+        content: el.content
+      });
+    });
+
+    //
+    _this.addEventsToDom();
```

Comment:
This trigger here is more of a callback. Would it nice to build a callback into this function rather than chaining? That way the call to this function could be seen in the `init`